### PR TITLE
Json changes

### PR DESCRIPTION
**Pull request by tonyma001**
Hi Trevor and Anum,
Please review the changes for HNSecure as well.

Thanks,
Tony

### CODE REVIEW

File: hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/FhirPayloadExtractor.java [hunk 1]
```diff
@@ -1,55 +1,33 @@
...
 import org.apache.camel.Handler;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.io.UnsupportedEncodingException;
-import java.nio.charset.StandardCharsets;
-import java.util.Base64;
+import ca.bc.gov.hlth.hnclientv2.Util;
+import ca.bc.gov.hlth.hnclientv2.json.FHIRJsonMessage;
+import ca.bc.gov.hlth.hnclientv2.json.FHIRJsonUtil;
```
Comment (tschia) [inline hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/FhirPayloadExtractor.java:11]: HNClient is a separate project (and deployed separately) we can't share dependencies between these projects. 
Can you add this functionality as classes to the HNI-ESB project? 

(Maybe in the future we could look at creating a library we can share as a dependency for both)

**Review by tschia — CHANGES_REQUESTED** [review]
Hi Tony,
See my comment about the imports from another project. We'll need to actually add these classes to this project.

File: hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/FhirPayloadExtractor.java (hunk 1 above)
Comment (pweckermann) [reply hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/FhirPayloadExtractor.java:11]: Agreed, I think we should consider splitting/sharing this component as part of a reference implementation. but that's something we can look into later.

Comment (tonyma001) [reply hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/FhirPayloadExtractor.java:11]: Thanks guys,
I will add this functionality into the HNI-ESB project and let you know.

Tony

Comment (tonyma001) [reply hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/FhirPayloadExtractor.java]: @tschia 
Hi Trevor, this functionality has been added into the HNI-ESB project and please review the code.

Thanks,
Tony

**Review by tschia — APPROVED** [review]
Approved

**Comment by tschia** [comment]
I made two small changes:
1.  I don't think we should pull in a large dependency like netty just for a single method StringUtil.isNullOrEmpty()
2. Lets try and use { } on all of our conditionals (if/else if/else) even if they're just one liners

This is approved, feel free to merge it.

**Comment by tonyma001** [comment]
Thanks Trevor. Agreed, your changes look good to me and I will try to merge them into the main branch then.

**Pull request merged**